panther_analysis_tool/lookup_table.py

```diff
@@ -0,0 +1,283 @@
+
+        self.lookup_id = response_payload['lookups'][0]['id']
+        return True
+
+
+    def create(self) -> int:
+        request_payload = json.dumps(
+            {"createLookup":
+             {"name": self.name, "enabled": self.spec["enabled"],
+              "description": self.spec.get("description", ""),
+              "reference": self.spec.get("reference", ""),
+              "lookupSchema": self.spec["lookupSchema"], "logTypeMap": self.spec["logTypeMap"]}}
+        )
+        logging.debug(request_payload)
+
+        response = self.client.invoke(
+            FunctionName=LOOKUP_LAMBDA_NAME,
+            InvocationType="RequestResponse",
+            Payload=request_payload,
+        )
+        response_str = response["Payload"].read().decode("utf-8")
+        response_payload = json.loads(response_str)
+        logging.debug(response_payload)
+
+        status_code = response["StatusCode"]
+        if status_code != 200:
+            logging.warning(
+                "Failed to search for the %s Lookup Table\n\tstatus code: %s\n\terror message: %s",
+                self.name,
+                status_code,
+                response_payload.get("errorMessage", response_payload.get("body")),
+            )
+            return 1
+        return 0
+
+
+    def update(self) -> int:
+        request_payload = json.dumps(
+            {"updateLookup":
+             {"name": self.name, "enabled": self.spec["enabled"],
+              "description": self.spec.get("description", ""),
+              "id": self.lookup_id, "reference": self.spec.get("reference", ""),
+              "lookupSchema": self.spec["lookupSchema"], "logTypeMap": self.spec["logTypeMap"]}}
+        )
+        logging.debug(request_payload)
+
+        response = self.client.invoke(
+            FunctionName=LOOKUP_LAMBDA_NAME,
+            InvocationType="RequestResponse",
+            Payload=request_payload,
+        )
+        response_str = response["Payload"].read().decode("utf-8")
+        response_payload = json.loads(response_str)
+        logging.debug(response_payload)
+        error_msg = response_payload.get("errorMessage", None)
+
+        status_code = response["StatusCode"]
+        if status_code != 200 or error_msg:
+            logging.warning(
+                "Failed to search for the %s Lookup Table\n\tstatus code: %s\n\terror message: %s",
+                self.name,
+                status_code,
+                response_payload.get("errorMessage", response_payload.get("body")),
+            )
+            return 1
+
+        if self.local_file_matches():
+            logging.info("The local data file matches S3 and does not need to be uploaded")
+        else:
+            logging.info("The local data file does not match S3 and needs to be uploaded.")
+            if self.upload_data():
+                return 1
+            if self.upload_from_s3():
+                return 1
+        return 0
+
+
+    def get_input_bucket(self) -> str:
```

Comment:
I suggest you save the value in a class member so the lambda only gets called once.

Reply:
Yeah, was going to set it via @property but it can raise an exception, which feels slightly odd, but maybe it's still Pythonic.

Reply:
going away